core/src/runtime/mod.rs

```diff
@@ -40,11 +41,16 @@ use tokio_timer::{timer, Interval};
 /// Supported Unix signals.
 #[derive(Copy, Clone, Debug)]
 pub enum UnixSignal {
+    /// Signal sent when controlling terminal is closed.
     SIGHUP = libc::SIGHUP as isize,
+    /// Signal sent to process for interruption, i.e. Ctrl+c.
     SIGINT = libc::SIGINT as isize,
+    /// Generic signal sent to terminal program.
     SIGTERM = libc::SIGTERM as isize,
 }
 
+/// Main runtime structure for handling ports, mempools, core-mapping, signals,
+/// and configuration.
```

Comment:
```
The Capsule runtime.

The runtime initializes the underlying DPDK environment. It also manages the task scheduler that executes the packet processing pipelines.
```

we will expand this more because this is the main struct applications will use. but we will save it for another day.